xkafka/config.go

```diff
@@ -0,0 +1,85 @@
+package xkafka
+
+import "time"
+
+// ConsumerConfig are configurations for creating a Kafka consumer.
+type ConsumerConfig struct {
+	Topics             []string      `yaml:"TOPICS" env:"TOPICS"`
+	Brokers            string        `yaml:"BROKERS" env:"BROKERS"`
+	Group              string        `yaml:"GROUP" env:"GROUP"`
+	TopicOffset        string        `yaml:"TOPIC_OFFSET" env:"TOPIC_OFFSET"`
+	AutoCommit         bool          `yaml:"AUTO_COMMIT" env:"AUTO_COMMIT"`
+	SessionTimeout     time.Duration `yaml:"SESSION_TIMEOUT" env:"SESSION_TIMEOUT"`
+	HeartbeatInterval  time.Duration `yaml:"HEARTBEAT_INTERVAL" env:"HEARTBEAT_INTERVAL"`
+	AutoCommitInterval time.Duration `yaml:"AUTO_COMMIT_INTERVAL" env:"AUTO_COMMIT_INTERVAL"`
+	SocketTimeout      time.Duration `yaml:"SOCKET_TIMEOUT" env:"SOCKET_TIMEOUT"`
+	ShutdownTimeout    time.Duration `yaml:"SHUTDOWN_TIMEOUT" env:"SHUTDOWN_TIMEOUT"`
+	PollTimeout        time.Duration `yaml:"POLL_TIMEOUT" env:"POLL_TIMEOUT"`
+	Concurrency        int           `yaml:"CONCURRENCY" env:"CONCURRENCY"`
+
+	// Used with BatchConsumer
+	BatchMaxSize int           `yaml:"MAX_SIZE" env:"MAX_SIZE"`
+	BatchMaxWait time.Duration `yaml:"MAX_WAIT" env:"MAX_WAIT"`
+
+	// Used as timeout while fetching metadata
+	MetadataRequestTimeout int `yaml:"METADATA_REQUEST_TIMEOUT" env:"METADATA_REQUEST_TIMEOUT"`
+}
+
+// SetDefaults sets default values for ConsumerConfig.
+func (c *ConsumerConfig) SetDefaults() {
```

Comment:
For managing config/behaviour, I have seen that `Option(s)` based API is more readily adopted in newer projects, ex: kube-api, open-telemetry, etc.
Using the same in [`courier-go`](https://github.com/gojek/courier-go/blob/5f1e90a9bea274dc818b8347fe9f4b1d75c341c6/client_options.go) as well.

It also [helps retain options type information](https://github.com/gojek/courier-go/blob/5f1e90a9bea274dc818b8347fe9f4b1d75c341c6/decoder.go#L30), which I think might not be required in this module.

Reply:
Option 1
```
consumer, err := xkafka.NewSimpleConsumer(
		"consumer-id",
		xkafka.WithTopics("test"),
		xkafka.WithBrokers("localhost:9092"),
		xkafka.WithKafkaConfig("group.id", "test"),
		xkafka.WithErrorHandler(ignoreError),
	)
```

Option 2
```
cfg := xkafka.ConsumerConfig{
		Topics:  []string{"test"},
		Brokers: "localhost:9092",
		Group:   "test",
	}

	consumer, err := xkafka.NewSimpleConsumer(
		"consumer-id",
		xkafka.WithConsumerConfig(cfg),
		xkafka.WithErrorHandler(ignoreError),
	)
```

I prefer option 1. we can solve config loading separately